mobile/ios/ios-device.ts

```diff
@@ -107,8 +126,126 @@ export class IOSDevice implements Mobile.IIOSDevice {
+			this.disconnect();
+		}
+	}
+
+	private findDeveloperDirectory(): IFuture<string> {
+		return (() => {
+			var childProcess = this.$childProcess.spawnFromEvent("xcode-select", ["-print-path"], "close").wait();
+			return childProcess.stdout.trim();
+		}).future<string>()();
+	}
+
+	private findDeveloperDiskImageDirectoryPath(): IFuture<string> {
+		return (() => {
+			var developerDirectory = this.findDeveloperDirectory().wait();
+			var buildVersion = this.getDeviceValue("BuildVersion");
+			var productVersion = this.getDeviceValue("ProductVersion");
+			var productVersionParts = productVersion.split(".");
+			var productMajorVersion = productVersionParts[0];
+			var productMinorVersion = productVersionParts[1];
+
+			var developerDiskImagePath = path.join(developerDirectory, "Platforms", "iPhoneOS.platform", "DeviceSupport");
+			var supportPaths = this.$fs.readDirectory(developerDiskImagePath).wait();
+
+			var supportPath: any = null;
+
+			_.each(supportPaths, (sp: string) => {
+				var parts = sp.split(' ');
+				var version = parts[0];
+				var versionParts = version.split(".");
+
+				var supportPathData = {
+					version: version,
+					majorVersion: versionParts[0],
+					minorVersion: versionParts[1],
+					build: parts.length > 1 ? parts[1].replace("(", "").replace(")", "") : null,
+					path: path.join(developerDiskImagePath, sp)
+				}
+
+				if(supportPathData.majorVersion === productMajorVersion) {
+					if(!supportPath) {
+						supportPath = supportPathData;
+					} else {
+						// is this better than the last match?
+						if(supportPathData.minorVersion === productMinorVersion) {
+							if(supportPathData.build === buildVersion) {
+								// perfect match
+								supportPath = supportPathData;
+							} else {
+								// we're still better than existing match
+								if(supportPath.build !== supportPathData.build || supportPath.build === null) {
+									supportPath = supportPathData;
+								}
+							}
+						}
+					}
+				}
+			});
+
+			if(!supportPath) {
+				this.$errors.fail("Unable to find device support path");
+			}
+
+			return supportPath.path;
+		}).future<string>()();
+	}
+
+	private mountImage(): void {
+		this.connect();
+		try {
+			this.startSession();
+			try {
+				var developerDiskImageDirectoryPath = this.findDeveloperDiskImageDirectoryPath().wait();
+				var imagePath = path.join(developerDiskImageDirectoryPath, "DeveloperDiskImage.dmg");
+				this.$logger.info("Mounting %s", imagePath);
+
+				var signature = this.$fs.readFile(util.format("%s.signature", imagePath)).wait();
+				var cfImagePath = this.$coreFoundation.createCFString(imagePath);
+
+				var cfOptions = this.$coreFoundation.cfTypeFrom({
+					ImageType: "Developer",
+					ImageSignature: signature
+				});
+
+				var result = this.$mobileDevice.deviceMountImage(this.devicePointer, cfImagePath, cfOptions, this.mountImageCallbackPtr);
+
+				if(result !== 0 && result !== 3892314230) { // 3892314230 - already mounted
```

Comment:
Considering that you already named this constant with a comment, perhaps we can make it official by introducing a constant? After the TypeScript 1.4 PR you can even literally use ```const```.